gobblin-utility/src/main/java/gobblin/util/PasswordUtils.java

```diff
@@ -0,0 +1,177 @@
+/*
+ * Copyright (C) 2014-2015 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use
+ * this file except in compliance with the License. You may obtain a copy of the
+ * License at  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed
+ * under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+ * CONDITIONS OF ANY KIND, either express or implied.
+ */
+
+package gobblin.util;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.net.URI;
+import java.util.Properties;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.hadoop.fs.Path;
+import org.jasypt.util.text.StrongTextEncryptor;
+
+import com.google.common.base.Preconditions;
+import com.google.common.base.Throwables;
+import com.google.common.io.Closer;
+import com.google.common.io.LineReader;
+
+import gobblin.configuration.ConfigurationKeys;
+import gobblin.configuration.State;
+
+/**
+ * A utility class for encrypting and decrypting passwords.
+ */
+public class PasswordUtils {
```

Comment:
The ENCRYPTOR object does not seem to be used.

I suggest considering a different design where encryptPassword/decryptPassword/readPassword methods are instance-level methods and PasswordUtils is an object with a encryptor/decryptor instance that gets initialized in the constructor . There can be different getInstance(...) methods which can read the master password from the specified location. Internally, getInstance() can use an instance cache keyed on the master password in case instantiating a new encryptor/decryptor is expensive.